packages/lib/src/memoize_timeout.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * Memoize a function with a timeout. Calling the function again after the timeout will erase the cache.
+ * @param fn Function to memoize
+ * @param time Time in ms to cache results
+ * @returns Memoized function
+ */
+// eslint-disable-next-line @typescript-eslint/ban-types
+export function memoizeTimeout(fn: Function, time: number): Function {
```

Comment:
This erases the the type information of the function passed into it. So for example if you do
```ts
function foo(input: string) {
    return input;
}
const memoized = memoizeTimeout(foo, 1000);
memoized(10, true);
```

Ther'es no type error as the function signature was lost. Instead we can declare a generic param extending `(...args: any) => any` to indicate it's a function (instead of the bad type Function), and use `Parameters<T>` and ReturnType<T> to create a new callable type with the same parameterns and return value. Note that it's not possible to use `T` as the returned value as that's making a promise that the returned type is exactly the same as the input type, which is not the case as a new function is returned that might not have the same properties as the input. See [this StackOverflow answer](https://stackoverflow.com/questions/69783310/type-is-assignable-to-the-constraint-of-type-t-but-t-could-be-instantiated) for an in depth explanation.

```suggestion
export function memoizeTimeout<T extends (...args: any) => any>(
	fn: T,
	time: number
): (...args: Parameters<T>) => ReturnType<T>
```